keras_rcnn/layers/object_detection/proposal_target.py

```diff
@@ -0,0 +1,60 @@
+import keras.backend
+import keras.engine
+import tensorflow
+
+import keras_rcnn.backend
+
+
+class ProposalTarget(keras.engine.topology.Layer):
+    def __init__(self, **kwargs):
+        self.allowed_border = 0
+        self.clobber_positives = False
+        self.negative_overlap = 0.3
+        self.positive_overlap = 0.7
+
+        super(ProposalTarget, self).__init__(**kwargs)
+
+    def build(self, input_shape):
+        super(ProposalTarget, self).build(input_shape)
+
+    def call(self, inputs, **kwargs):
+        scores, gt_boxes, im_info = inputs
+
+        # TODO: Fix usage of batch index
+        shape = im_info[0, :2]
+        scale = im_info[0, 2]
+
+        # 1. Generate proposals from bbox deltas and shifted anchors
+        all_anchors = keras_rcnn.backend.shift(shape, 16)
+
+        # only keep anchors inside the image
+        indices, anchors = keras_rcnn.backend.inside_image(all_anchors, im_info, self.allowed_border)
+
+        # label: 1 is positive, 0 is negative, -1 is dont care
+        ones   = keras.backend.ones_like(indices, dtype=keras.backend.floatx())
+        zeros  = keras.backend.zeros_like(indices, dtype=keras.backend.floatx())
+        labels = ones * -1
+
+        argmax_overlaps_inds, max_overlaps, gt_argmax_overlaps_inds = keras_rcnn.backend.overlapping(anchors, gt_boxes, indices)
+
+        if not self.clobber_positives:
+            labels = keras_rcnn.backend.where(keras.backend.less(max_overlaps, self.negative_overlap), zeros, labels)
+
+        # fg label: for each gt, anchor with highest overlap
+        #TODO Find Keras equivalent of this
+        #labels[gt_argmax_overlaps_inds] = 1
```

Comment:
You could use something like:

`keras_rcnn.backend.scatter_add_tensor(labels, indices, keras.backend.ones((...)))`